packages/react-theming/src/compose.ts

```diff
@@ -130,20 +136,39 @@ const _mergeOptions = (options: ComposeOptions, baseOptions?: Options): Options
   return optionsSet;
 };
 
+/**
+ * _tokensFromOptions returns the accurate set of tokens
+ * based on the currently rendered props.
+ *
+ * @internal
+ *
+ * @param options A set of options
+ * @param props Props for this render
+ */
+export const _tokensFromOptions = (options: any[], props: any) => {
+  let result = options.reduce((acc, option) => {
+    return { ...acc, ...(option.tokens || {}) };
+  }, {});
+  options.forEach(option => {
+    Object.keys(option.variants || {}).forEach(variantName => {
+      const v: Variant = option.variants[variantName];
+      result = { ...result, ...v.tokens(props[variantName]) };
+    });
+  });
+  return result;
+};
+
 const _getClasses = (
   name: string | undefined,
   theme: ITheme,
   classNamesCache: WeakMap<any, any>,
   optionsSet: any[],
+  props: any,
 ) => {
-  let classes = classNamesCache.get(theme);
+  let classes = null; // classNamesCache.get(theme);
```

Comment:
I don't think this was intentional.

Reply:
For now it is, as this PR doesn't yet implement correct caching based on props. Will be in a following PR once we ratify the design of variants. Good callout though!